daemon/daemon.go

```diff
@@ -407,20 +407,12 @@ func (daemon *Daemon) reserveName(id, name string) (string, error) {
 
 		conflictingContainer, err := daemon.GetByName(name)
 		if err != nil {
-			if strings.Contains(err.Error(), "Could not find entity") {
-				return "", err
-			}
```

Comment:
Thanks @mountkin, and thanks for finding this situation; we were scratching our heads a bit because of this code, and it looks like there are unhandled situations :+1: